ssas/service/blacklist/blacklist.go

```diff
@@ -0,0 +1,325 @@
+package blacklist
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"regexp"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/CMSgov/bcda-ssas-app/ssas"
+	"github.com/CMSgov/bcda-ssas-app/ssas/cfg"
+	"github.com/patrickmn/go-cache"
+	"github.com/pborman/uuid"
+)
+
+// Singleton instance so we can share a common blacklist instance across various SSAS modules
+var Blacklist *blacklist
+
+// This function should only be called by main
+func Start() {
+	// Guarantee that we only have a single Blacklist instance running
+	if Blacklist != nil {
+		return
+	}
+	Blacklist = newBlacklist(24*time.Hour,
+		time.Duration(cfg.GetEnvInt("SSAS_TOKEN_BLACKLIST_CACHE_CLEANUP_MINUTES", 15))*time.Minute,
+		time.Duration(cfg.GetEnvInt("SSAS_GROUP_BLACKLIST_CACHE_CLEANUP_MINUTES", 15))*time.Minute,
+		time.Duration(cfg.GetEnvInt("SSAS_TOKEN_BLACKLIST_CACHE_TIMEOUT_MINUTES", 60*24))*time.Minute,
+		time.Duration(cfg.GetEnvInt("SSAS_GROUP_BLACKLIST_CACHE_TIMEOUT_MINUTES", 60*24))*time.Minute,
```

Comment:
I may not be understanding the use case, but I'm surprised that we're only blacklisting an org for 24 hours.

Reply:
Good call - I think we'll look at requiring the fields to be defined by the application (and not provide a default value).